### PR TITLE
detect/byte_jump: Implement "from_end" option

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #4947 

This PR adds support for `from_end` to the existing `byte_jump`. Although documented, it was not implemented.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [3626](https://redmine.openinfosecfoundation.org/issues/3626)

Describe changes:
- Address review comments.

Companion [Suricata-verify PR #232](https://github.com/OISF/suricata-verify/pull/232)

### CODE REVIEW

File: src/detect-bytejump.c [hunk 1]
```diff
@@ -167,12 +167,14 @@ int DetectBytejumpDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s,
     /* Calculate the jump location */
     if (flags & DETECT_BYTEJUMP_BEGIN) {
         jumpptr = payload + val;
-        //printf("NEWVAL: payload %p + %ld = %p\n", p->payload, val, jumpptr);
-    }
-    else {
+        SCLogDebug("NEWVAL: payload %p + %" PRIu64 "= %p\n", payload, val, jumpptr);
+    } else if (flags & DETECT_BYTEJUMP_END) {
+        jumpptr = payload + payload_len + val;
+        SCLogDebug("NEWVAL: payload %p + %" PRIu64 " - %" PRIu64 " = %p\n", payload, payload_len, val, jumpptr);
```
Comment (victorjulien) [inline src/detect-bytejump.c:173]: payload_len is uint32_t (see travis failure)

**Comment by victorjulien** [comment]
Didn't see a response to this comment from the last PR: "Is it possible to specify both from_beginning and from_end? If so, that should probably invalidate the sig? Did you try that?"

**Comment by jlucovsky** [comment]
> Didn't see a response to this comment from the last PR: "Is it possible to specify both from_beginning and from_end? If so, that should probably invalidate the sig? Did you try that?"

Good point. I'll add a check and a unittest.

**Comment by jlucovsky** [comment]
Continued in #4952

**Pull request closed without merging**